Odata-docs/webapi-8/fundamentals/query-options.md

```diff
@@ -0,0 +1,486 @@
+
+A query option is basically requesting that a service perform a set of transformations such as filtering, sorting, etc. to its data before returning the results. See more information [here](/odata/concepts/queryoptions-overview).
+
+## Model class
+We will define our model as follows:
+
+```csharp
+public class Customer
+{
+    [Key]
+    public int Id { get; set; }
+    public String Name { get; set; }
+    public int Age { get; set; }
+    public List<Order> Orders { get; set; }
+}
+
+public class Order
+{
+    [Key]
+    public int Id { get; set; }
+    public int Price { get; set; }
+    public int Quantity { get; set; }
+}
+```
+
+We have a `Customer` entity with `Orders` navigation property.
+
+## Configuring Startup class
+```csharp
+public class Startup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        var modelBuilder = new ODataConventionModelBuilder();
+        modelBuilder.EntitySet<Customer>("Customers");
+
+        services.AddControllers().AddOData(
+            options => options.Select().Filter().OrderBy().Expand().Count().SetMaxTop(null).AddRouteComponents(
+                "odata",
+                modelBuilder.GetEdmModel()));
+    }
+}
+```
+
+In the above configuration, `options` is an instance of `ODataOptions`.
+`Select()` enables `$select` query option.
+`Filter()` enables `$filter` query option.
+`OrderBy()` enables `$orderby` query option.
+`Expand()` enables `$expand` query option.
+`Count()` enables `$count` query option.
+`SetMaxTop()` sets the maximum value of `$top` that a client can request.
+
+## Controller
+
+```csharp
+public class CustomersController : ODataController
+{
+    [EnableQuery]
+    public IActionResult Get()
+    {
+        return Ok(context.Customers);
+    }
+}
+```
+
+In the controller method, we add the `EnableQuery` attribute.
+The `EnableQuery` attribute is an action filter that parses, validates, and applies the query. The filter converts the query options into a LINQ expression. When the controller returns an `IQueryable` or `IActionFilter` type, the Linq provider converts the Linq expression into a query e.g EfCore will convert the Linq expression to a Sql statement. 
+
+## Basic queries
+We can select a specific property or properties in an entity.
+
+```http
+GET http://localhost:6285/odata/Customers?$select=Name
+```
+
+```json
+{
+    "@odata.context": "http://localhost:6285/odata/$metadata#Customers(Name)",
+    "value": [
+        {
+            "Name": "Customer 1"
+        },
+        {
+            "Name": "Customer 2"
+        },
+        {
+            "Name": "Customer 3"
+        },
+        {
+            "Name": "Customer 4"
+        },
+        {
+            "Name": "Customer 5"
+        }
+    ]
+}
+```
+
+We can return `Customers` with their related `Orders`. In this case we use `$expand`.
+
+```http
+GET http://localhost:6285/odata/Customers?$expand=Orders
+```
+
+```json
```

Comment:
To keep the size of you doc manageable, consider having 3 objects in your data source. They should be enough to help you demonstrate application of OData query options